src/bucket.rs

```diff
@@ -197,6 +197,40 @@ impl Bucket {
                 Alignment::Align64 => 16,
             }
     }
+
+    // remove an element - we assume there's an element
+    pub fn remove(&mut self, offset: usize) -> BucketElement {
+        let elem = self.tab[offset];
+        let len = self.tab.len();
+
+        // remove element from table
+        self.tab[offset] = BucketElement::default();
+        self.count -= 1;
+
+        // Compile a list of all elements between the removed element and
+        // the next empty one, that aren't in their home position.
+        let possible_moves = (1..len)
+            .map(|i| offset + i)
+            .take_while(|offset| self.tab[offset % len].is_occupied())
+            .filter_map(|offset| {
+                let index = offset % len;
+                let home = self.tab[index].hash as usize % len;
+                (index != home).then_some((offset, home))
+            })
+            .collect::<Vec<_>>();
+
+        // Move any of the out-of-place elements downwards into the
+        // hole if the hole is between them and their home.
+        let mut hole = offset;
+        possible_moves.into_iter().for_each(|(offset, home)| {
+            if hole >= home {
+                self.tab.swap(offset % len, hole % len);
+                hole = offset;
+            }
```

Comment:
It is perfectly safe to swap. We initialize `hole` with an empty element. We then swap that with the element we want to move. Then point hole to the empty element. I even added a test, because this is the most awkward piece of code in the whole library.

For future, I'd recommend simplifying the in-memory representation to a `<vec<Vec<u64>>`. Then remove is simply a `Vec::remove()` inset is `Vec::insert()`.